### PR TITLE
docs: features & working of the project

### PR DESCRIPTION
**Pull request by ankithans**
links to #1 

Initial Planning of features and implementation

### CODE REVIEW

**Comment by wtrocki** [comment]
I have added more info but this needs more work

**Comment by ankithans** [comment]
okay, I am on it!

**Comment by wtrocki** [comment]
Pushed more changes

**Comment by ankithans** [comment]
https://ankithans.github.io/charmil/
added some more commands and architecture diag

File: README.md [hunk 1]
```diff
@@ -1 +1,44 @@
...
+
+`$ charmil extensions list`
+
+`$ charmil extensions installed` # list installed ones
+
+`$ charmil extensions install <name>@<version>`
+
+`$ charmil extensions remove <name>@<version>`
+
+`$ charmil extentions <name>@<version> <commands>` # installed extensions will become child of rootCLI (need review)
```
Comment (wtrocki) [inline README.md:26]:
```suggestion

```

**Comment by wtrocki** [comment]
I really do not get what `init command` would do. 
Also readme still missing tons of info that we can add later.

**Pull request merged**